test/hotspot/jtreg/compiler/conversions/TestMoveConvI2LThroughAddIs.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.conversions;
+
+import jdk.test.lib.Asserts;
+
+/*
+ * @test
+ * @bug 8254317
+ * @requires vm.compiler2.enabled
+ * @summary Exercises the optimization that moves integer-to-long conversions
+ *          upwards through different shapes of integer addition
+ *          subgraphs. Contains three basic (small) tests and two stress tests
+ *          that resulted in a compilation time and memory explosion, triggering
+ *          the short specified timeout.
+ * @library /test/lib /
+ * @run main/othervm -Xcomp -XX:-TieredCompilation -XX:-Inline
```

Comment:
For future tests writing.
In general we should avoid using -Xcomp because it does not provide profiling information to C2 and it may generate unexpected code because branches frequencies are not known.
Preferable way is warm up test method by calling it in a loop enough times to make sure it is compiled and then verify results outside test method.
Then you don't need -XX:CompileOnly command.

Reply:
Without -Xcomp, the final code would not the same as expected, lots of branches have been optimized with profiling data which we **do** normally used. 

I wonder whether it's feasible to create such a test case that grows exponentially with profiling data. I think that may decriable the issue more realistic.

Reply:
Thanks for the feedback! The goal of these test cases is to exercise the logic in `ConvI2LNode::Ideal()` in a way as isolated as possible from the rest of the JVM, for simplicity, reproducibility, and ease of debugging. I will rewrite them to verify the results outside the test method. I will see if I can get rid of `-Xcomp` and `-XX:CompileOnly` without making them too complex. If that does not work, I will try to construct another test case as suggested by @erik1iu .

Reply:
I have addressed this in the latest update.